app/components/Search/mazemapAutocomplete.js

```diff
@@ -51,67 +41,64 @@ function mazemapAutocomplete<Props>({
     state = {
       searching: false,
       result: [], //first result fetched on mount
+      Mazemap: undefined,
     };
 
     _isMounted = false;
 
     componentDidMount() {
+      import('mazemap').then((mazemap) => this.setState({ Mazemap: mazemap }));
       this._isMounted = true;
-      if (this.props.meta.initial) {
-        Mazemap.Data.getPoi(this.props.meta.initial.value).then((poi) =>
-          this.setState({
-            result: [
-              mapRoomAndBuildingToResult(
-                poi.properties.title,
-                poi.properties.buildingName,
-                this.props.meta.initial.value
-              ),
-            ],
-          })
-        );
-      }
     }
 
     componentWillUnmount() {
       this._isMounted = false;
     }
 
-    handleSearch = (query: string): void => {
-      if (!query) {
-        return;
-      }
-      this.setState({
-        searching: true,
-      });
-
-      mazemapSearchController
-        .search(query)
-        .then((results) => {
-          if (this._isMounted) {
-            this.setState({
-              result: results.results.features.map((result) =>
-                mapRoomAndBuildingToResult(
-                  result.properties.dispPoiNames[0],
-                  result.properties.dispBldNames[0],
-                  result.properties.poiId
-                )
-              ),
-            });
-          }
-        })
-        .finally(() => {
-          if (this._isMounted) {
-            this.setState({ searching: false });
-          }
-        });
-    };
-
     render() {
+      const { Mazemap } = this.state;
+      const handleSearch = (query: string): void => {
+        if (!query || !Mazemap) {
+          return;
+        }
+        const mazemapSearchController = new Mazemap.Search.SearchController({
+          campusid: 1,
+          rows: 10,
+          withpois: true,
+          withbuilding: false,
+          withtype: false,
+          withcampus: false,
+          resultsFormat: 'geojson',
+        });
+        this.setState({
+          searching: true,
+        });
+        mazemapSearchController
+          .search(query)
+          .then((results) => {
+            if (this._isMounted) {
+              this.setState({
+                result: results.results.features.map((result) =>
+                  mapRoomAndBuildingToResult(
+                    result.properties.dispPoiNames[0],
+                    result.properties.dispBldNames[0],
+                    result.properties.poiId
+                  )
+                ),
+              });
+            }
+          })
+          .finally(() => {
+            if (this._isMounted) {
+              this.setState({ searching: false });
+            }
+          });
+      };
```

Comment:
Hmm, why did you move these methods into `render()`? This will make every render heavier as it sets up new functions for every render and not when creating the component as before.

Also, it's probably alot better to initialize the `Mazemap.Search.SearchController` only once in the mount when you do the import, and rather store that in state instead of the entire Mazemap lib, since it seems that that's the only thing used here.

Reply:
Good questions all around, I've done some cleanup and moved the methods back out of render(), in what is hopefully the final™ commit on this pr.